packages/strapi-utils/lib/build-query.js

```diff
@@ -110,33 +110,41 @@ const hasDeepFilters = ({ where = [], sort = [] }, { minDepth = 1 } = {}) => {
 };
 
 const normalizeWhereClauses = (whereClauses, { model }) => {
-  return whereClauses
-    .filter(({ value }) => !_.isNil(value))
-    .map(({ field, operator, value }) => {
-      if (BOOLEAN_OPERATORS.includes(operator)) {
-        return {
-          field,
-          operator,
-          value: value.map(clauses => normalizeWhereClauses(clauses, { model })),
-        };
-      }
-
-      const { model: assocModel, attribute } = getAssociationFromFieldKey({
-        model,
-        field,
-      });
-
-      const { type } = _.get(assocModel, ['allAttributes', attribute], {});
-
-      // cast value or array of values
-      const castedValue = castInput({ type, operator, value });
+  return whereClauses.map(({ field, operator, value }) => {
+    if (_.isNil(value)) {
```

Comment:
Calling the db with a null should be valid & the db layer will actually make the filtering. We need to be able to find an entry with a value equal to null :)

Reply:
Checking if you will have time to finish this :)

Reply:
Alex before you close the PR let me poke him on Slack, he has been a little busy lately ;)

Reply:
Sorry guys, yeah I had it quite busy last couple of weeks.

I update the PR so it only throws an error on `undefined` values, it skips `null` values as it did before.